src/analysis/processing/qgsalgorithmlineintersection.cpp

```diff
@@ -196,24 +196,44 @@ QVariantMap QgsLineIntersectionAlgorithm::processAlgorithm( const QVariantMap &p
           {
             outAttributes.append( inFeatureB.attribute( b ) );
           }
-          if ( intersectGeom.type() == QgsWkbTypes::PointGeometry )
+          if ( intersectGeom.wkbType() == QgsWkbTypes::GeometryCollection )
```

Comment:
Better check QgsWkbTypes::isMultiType( intersectGeom.wkbType()) here instead - wkbType may be MultiPoint, MultiLineString, GeometryCollectionZ, etc...

Reply:
The collection is a mixture of geometries. In the test data the intersection is a line and three points. We don't really want to catch here MultiGeometries but collections. 

Having said that, I think the best option is `QgsWkbTypes.flatType(intersectGeom.type()) == QgsWkbTypes.GeometryCollection`. Do you agree?